virtual_rainforest/main.py

```diff
@@ -171,6 +173,85 @@ def extract_timing_details(
     return start_time, update_interval, raw_interval, end_time
 
 
+def output_current_state(
+    data: Data,
+    models_cfd: dict[str, BaseModel],
+    data_options: dict[str, Any],
+    time_index: int,
+) -> None:
+    """Function to output the current state of the data object.
+
+    This function outputs all variables stored in the data object, except for any data
+    with a "time_index" dimension defined (at present only climate input data has this).
+    This data can either be saved as a new file or appended to an existing file.
+
+    Args:
+        data: Data object to output current state of
+        models_cfd: Set of models configured for use in the simulation
+        data_options: Set of options concerning what to output and where
+        time_index: The index representing the current time step in the data object.
+
+    Raises:
+        ConfigurationError: If the final output directory doesn't exist, isn't a
+           directory, or the final output file already exists (when in new file mode).
+           If the file to append to is missing (when not in new file mode).
+    """
+
+    # Only variables in the data object that are updated by a model should be outputted
+    all_variables = [
+        models_cfd[model_nm].vars_updated for model_nm in models_cfd.keys()
+    ]
+    # Then flatten the list
+    variables_to_save = [item for sublist in all_variables for item in sublist]
+    # Create output file path for specific time index
+    out_path_name = (
+        f"{data_options['out_folder_continuous']}/continuous_state{time_index}.nc"
```

Comment:
I would pad the `time_index` with zeros, so the files are ordered. I know they are going to be deleted, but it is a question of principles :) . For example, to write the index in a field with 5 digits, you would do:

```suggestion
        f"{data_options['out_folder_continuous']}/continuous_state{time_index:05}.nc"
```
Resulting in filenames:
```output
continuous_state00001.nc
continuous_state00002.nc
continuous_state00003.nc
...
continuous_state00345.nc
```